File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.cs
public class Solution {
    public int MinCostClimbingStairs(int[] cost) {
        int n = cost.Length;
        int[] dp = new int[n + 1];
        dp[0] = 0;
        dp[1] = 0;
        
         for (int i = 2; i <= n; i++) {
            int oneStep = dp[i - 1] + cost[i - 1];
            int twoSteps = dp[i - 2] + cost[i - 2];
            dp[i] = Math.Min(oneStep, twoSteps);
        }

        return dp[n];
    }
}

File: 747-min-cost-climbing-stairs/min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        
        int prev1 = cost[n-1];
        int prev2 = 0;

        for(int i = n - 2; i >= 0; i--) {
            int temp = cost[i] + min(prev1, prev2);
            prev2 = prev1;
            prev1 = temp;
        }

        return min(prev2, prev1);
    }
};